.github/scripts/filter_test_configs.py

```diff
@@ -191,12 +200,86 @@ def set_periodic_modes(
     return scheduled_test_matrix
 
 
+def mark_unstable_jobs(
+    workflow: str, job_name: str, test_matrix: Dict[str, List[Any]]
+) -> Dict[str, List[Any]]:
+    """
+    Check the list of unstable jobs and mark them accordingly. Note that if a job
+    is unstable, all its dependents will also be marked accordingly
+    """
+    return process_jobs(
+        workflow=workflow,
+        job_name=job_name,
+        test_matrix=test_matrix,
+        issue_type=IssueType.UNSTABLE,
+        url=UNSTABLE_JOBS_URL,
+    )
+
+
 def remove_disabled_jobs(
     workflow: str, job_name: str, test_matrix: Dict[str, List[Any]]
 ) -> Dict[str, List[Any]]:
     """
     Check the list of disabled jobs, remove the current job and all its dependents
-    if it exists in the list. The list of disabled jobs is as follows:
+    if it exists in the list
+    """
+    return process_jobs(
+        workflow=workflow,
+        job_name=job_name,
+        test_matrix=test_matrix,
+        issue_type=IssueType.DISABLED,
+        url=DISABLED_JOBS_URL,
+    )
+
+
+def _filter_jobs(
+    test_matrix: Dict[str, List[Any]],
+    issue_type: IssueType,
+    target_cfg: Optional[str] = None,
+) -> Dict[str, List[Any]]:
+    """
+    An utility function used to actually apply the job filter
+    """
+    # The result will be stored here
+    filtered_test_matrix: Dict[str, List[Any]] = {"include": []}
+
+    # This is an issue to disable a CI job
+    if issue_type == IssueType.DISABLED:
+        # If there is a target config, disable (remove) only that
+        if target_cfg:
+            # Remove the target config from the test matrix
+            filtered_test_matrix["include"] = [
+                r for r in test_matrix["include"] if r.get("config", "") != target_cfg
+            ]
```

Comment:
Do you need a line setting `r[IssueType.DISABLE.value] = IssueType.DISABLE.value`, similar to what UNSTABLE values have below?

Can you please explain why you're tracking the Unstable status? And why the Disabled status might not need to be tracked?

Reply:
This is not for disabled tests as the entry for a disabled job is removed and the job isn't even run.  For example, if I have the following test matrix:

```
{ include: [
  { config: "default", shard: 1, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 2, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 3, num_shards: 3, runner: "linux.2xlarge" },
  { config: "distributed", shard: 1, num_shards: 2, runner: "linux.2xlarge" },
  { config: "distributed", shard: 2, num_shards: 2, runner: "linux.2xlarge" },
]}
```

where distributed tests are disabled.  The filtered matrix will become:

```
{ include: [
  { config: "default", shard: 1, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 2, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 3, num_shards: 3, runner: "linux.2xlarge" },
]}
```

On the other hand, if distributed tests are marked as unstable, they would still be run and the filtered matrix will be:

```
{ include: [
  { config: "default", shard: 1, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 2, num_shards: 3, runner: "linux.2xlarge" },
  { config: "default", shard: 3, num_shards: 3, runner: "linux.2xlarge" },
  { config: "distributed", shard: 1, num_shards: 2, runner: "linux.2xlarge", issue_type: "unstable" },
  { config: "distributed", shard: 2, num_shards: 2, runner: "linux.2xlarge", issue_type: "unstable" },
]}
```